code/__DEFINES/~skyrat_defines/combat.dm

```diff
@@ -108,73 +108,84 @@
 #define HEADSMASH_BLOCK_ARMOR 20
 #define SUPLEX_TIMER 3 SECONDS
 
-//TODO: Add a grab state check on the do_afters
-/datum/species/proc/try_grab_maneuver(mob/living/carbon/human/user, mob/living/carbon/human/target, modifiers)
-	var/obj/item/bodypart/affecting = target.get_bodypart(ran_zone(user.zone_selected))
+/// Skyrat change - alt-clicking a human as another human while grappling them tightly makes you try for grappling-based maneuvers.
+/mob/living/carbon/human/AltClick(mob/user)
+	if(ishuman(user))
+		var/mob/living/carbon/human/human_user = user
+		if(human_user != src && human_user.combat_mode && human_user.dna.species.try_grab_maneuver(user, src))
+			return
+	. = ..()
+
+/// State check for grab maneuver - because you can't logically suplex a man if you've stopped grappling them.
+/datum/species/proc/grab_maneuver_state_check(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	return (target.pulledby && target.pulledby == user && user.grab_state > GRAB_PASSIVE)
+
+/// Tries a grab maneuver - suplex, limb dislocation, or headslam depending on targeted limb.
+/datum/species/proc/try_grab_maneuver(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	if(!grab_maneuver_state_check(user, target))
+		return
+	// psst, future coder - if you're adding more precise interactions, e.g. eye gouging/strangling, you're gonna need to make this less precise!
+	// just remove the deprecise_zone() call. account for the specifics after!
+	var/obj/item/bodypart/affecting = target.get_bodypart(deprecise_zone(user.zone_selected))
 	if(!affecting)
 		return FALSE
 	. = FALSE
```

Comment:
Could this be turned into a normal variable? If so, it would be more readable.

Reply:
i'm pretty sure that weird dot's also returned for reporting the success/failure of the proc, but then again a lot of other things also handle that. didn't touch it but i'll probably think about going back on that sooner or later